x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDataFrameAnalyticsAction.java

```diff
@@ -377,6 +377,9 @@ private void validateSourceIndexHasRows(StartContext startContext, ActionListene
                             startContext.config.getId(),
                             Strings.arrayToCommaDelimitedString(startContext.config.getSource().getIndex())
                         ));
+                    } else if (Math.floor(startContext.config.getAnalysis().getTrainingPercent() * dataSummary.rows)  >= Math.pow(2, 32)) {
```

Comment:
nit, there is no reason to calculate 2^32 here. I would just create a static variable where it is calculated once and thats it.